src/Api.js

```diff
@@ -33,6 +33,31 @@ class Api {
 		] );
 	}
 
+	/**
+	 * Get parsed edit summary for the given revision.
+	 * @param {number} revId
+	 * @return {Promise} Resolving Object with keys 'comment' and 'size'.
+ 	 */
+	fetchEditSummary( revId ) {
+		return this.mwApi.ajax( {
+			action: 'compare',
+			fromrev: revId,
+			torelative: 'prev',
+			prop: 'parsedcomment|size',
+			formatversion: 2
+		} ).then( data => {
+			if ( data.compare ) {
+				return $.Deferred().resolve( {
```

Comment:
You don't have to wrap this with a .resolve, because you're aleady in a resolved path (you do this when you want to flip the promise from resolve to reject and vise versa like you do below on line 55)

Just return the object here directly.

Reply:
Done!